### ISSUE TITLE
missing TERM_PROGRAM and TERM_PROGRAM_VERSION environment variables when launched via window manager, Dock, or launcher 

### ISSUE THREAD

**Issue by networkhermit** [zed-industries/zed#21951]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I launch Zed via system launcher and open a project, the login shell exited with exit status 1, the LSP path lookup also fails.

This is partly due to my zshrc automatically exec tmux upon startup, but Zed already set TERM_PROGRAM and TERM_PROGRAM_VERSION environment variables in integrated terminal (see #4571 and #14213 ), it would be more consistent if we could provide these environment variables to login shell too.

```bash
if [[ -z "${TMUX}" ]] && [[ -z "${VIM}" ]]; then
    if [[ "${TERM}" = xterm-256color ]] || [[ "${TERM}" = tmux-256color ]]; then
        case ${TERM_PROGRAM} in
        vscode | zed)
            ;;
        *)
            exec tmux new-session -A -D -s main
            ;;
        esac
    fi
fi
```

### Environment

Zed: v0.165.4 (Zed)
OS: macOS 15.2.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```
2024-12-13T11:06:35.707747+08:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-12-13T11:06:35.801047+08:00 [ERROR] login shell exited with exit status: 1
```
</details>

**Comment by notpeter** [comment]
Sorry for the delayed response. I don't understand what you are asking for here. Zed sets these environment variables correctly for me, both when I launch via the CLI or when I launch via the Dock/Finder.

```
echo $TERM_PROGRAM $TERM_PROGRAM_VERSION
```

If you have a shell that auto-launches tmux, tmux is setting it's own variables which occur because tmux is now the terminal emulator, not Zed (it handles window size, scrollback, cursor movement, etc)

**Comment by networkhermit** [comment]
@notpeter 

> If you have a shell that auto-launches tmux...

No. My shell would not auto-launch tmux when it detects the TERM_PROGRAM is zed.

> Zed sets these environment variables correctly for me, both when I launch via the CLI or when I launch via the Dock/Finder.

When launch zed via Dock/Finder **and open the built-in terminal**, `$TERM_PROGRAM` and `$TERM_PROGRAM_VERSION` are correctly set.

But that's not true for the main zed process or maybe LSP initialization process (I have used gopls and rust-analyzer), so the Zed.log exit with error as described (because the expected `TERM_PROGRAM` and `TERM_PROGRAM_VERSION` is missing):

```text
2024-12-13T11:06:35.801047+08:00 [ERROR] login shell exited with exit status: 1
```

Actually I think this is a regression, but I have not got the time to verify it.

Also, vscode works fine with this usage.

**Comment by notpeter** [comment]
I would not expect Zed to set TERM_PROGRAM when piping stdin/stdout to/from a subprocess like a language server. There literally is not a terminal program attached so anything which have term-specific behavior/workarounds should default to the dumbest behavior and assume this is like a dumb tty.

I understand it appears vscode sets this and you are looking for the ability to detect zed in your shell scripts, but I'm not convinced this is the appropriate method.

Happy to revisit if there's a spec/standard which suggested we should set this, but I believe only `TERM` is POSIX and TERM_PROGRAM and TERM_PROGRAM_VERSION is just by-convention.

**Comment by networkhermit** [comment]
> Happy to revisit if there's a spec/standard which suggested we should set this, but I believe only TERM is POSIX and TERM_PROGRAM and TERM_PROGRAM_VERSION is just by-convention.
>
> I understand it appears vscode sets this and you are looking for the ability to detect zed in your shell scripts, but I'm not convinced this is the appropriate method.

Yes. There is no formal standard for TERM_PROGRAM and TERM_PROGRAM_VERSION.

The TERM_PROGRAM and TERM_PROGRAM_VERSION is implemented correctly in zed and my unchanged setup works since a long time. I have no glue when this regression is introduced, might be months ago because I don't use zed in a daily basis and especially after it's broken for me and I have to troubleshoot the issue.

So here is the current situation if there is no plan to fix it: the TERM_PROGRAM and TERM_PROGRAM_VERSION is broken and semi-deprecated in Zed, don't rely on it anymore, there is no way to detect zed in shell scripts at the moment.
